controller/controller_test.go

```diff
@@ -94,17 +94,19 @@ func TestRunOnce(t *testing.T) {
 			Target:  "1.2.3.4",
 		},
 		{
-			DNSName: "update-record",
-			Target:  "8.8.4.4",
+			DNSName:   "update-record",
+			Target:    "8.8.4.4",
+			RecordTTL: 100,
 		},
 	}, nil)
 
 	// Fake some existing records in our DNS provider and validate some desired changes.
 	provider := newMockProvider(
 		[]*endpoint.Endpoint{
 			{
-				DNSName: "update-record",
-				Target:  "8.8.8.8",
+				DNSName:   "update-record",
+				Target:    "8.8.4.4",
+				RecordTTL: 50,
```

Comment:
Does this test only work with TTL?
I think we should have a default defined in the zone, similar to DNS origin definition to default TTLs.

If it works, can we make this testcase to have with and without TTL tests

Reply:
The TTLs are there in some tests now to have record updates rather than creations/deletions. Basically, with one Endpoint per (DNSName, Target), to update a record rather than delete it and create another one, you must not change either DNSName or Target, so I'm changing the TTL instead. If you change the target, the record won't be updated, it will be deleted and another one with the new target created -- similar to what would happen if you change the name. I have modified Plan.Compute() to operate this way. (that's really the crux of this PR)

Reply:
ok thanks to clarify this